### PR TITLE
Storage tweaks

### PR DESCRIPTION
**Pull request by daviddahl**
A number of enhancements around storage data format and logging - still in progress

### CODE REVIEW

File: lib/main.js [hunk 1]
```diff
@@ -316,3 +487,76 @@ function checkForPrivateChannel(aChannel) {
...
+    }
+
+    log("prePath: " + aHttpChannel.URI.prePath);
+    log("spec: " + aHttpChannel.URI.spec);
+    let prePath = aHttpChannel.URI.prePath;
+    let path = aHttpChannel.URI.path;
+    let url = prePath + path;
+    log("url: " + url);
+
+    if (kSHARE_URLS.indexOf(url)) {
```
Comment (monicachew) [inline lib/main.js:554]: If you strip the scheme here we don't have to include http[s] on the list

File: lib/main.js [hunk 2]
```diff
@@ -41,6 +74,20 @@ function log() {
...
+function pprint(aObj) {
+  if (!kDEBUG) {
+    return;
+  }
+  if (typeof aObj == "string") {
+    log(aObj);
+  }
+  else {
+    for (let prop in aObj) {
+      log(prop + ": " + aObj[prop]);
```
Comment (monicachew) [inline lib/main.js:96]: is this different than JSON.stringify?

File: lib/main.js [hunk 3]
```diff
@@ -41,6 +74,20 @@ function log() {
   }
 }
 
+function pprint(aObj) {
+  if (!kDEBUG) {
```
Comment (monicachew) [inline lib/main.js:88]: I learned about simple-prefs recently, from reading micropilot code (micropilotlog) that might be useful here

File: lib/main.js [hunk 4]
```diff
@@ -316,3 +487,76 @@ function checkForPrivateChannel(aChannel) {
...
   return false;
 }
+
+/**
+ * NetworkListener
+ *
+ * We need to check all HTTP traffic for social widgets
+ *
+ * https://mxr.mozilla.org/mozilla-central/source/netwerk/protocol/http/nsIHttpActivityObserver.idl#14
+ **/
```
Comment (monicachew) [inline lib/main.js:514]: Cool, thanks for finding this!

File: lib/main.js [hunk 5]
```diff
@@ -180,12 +230,48 @@ function parseCookie(aCookie) {
...
+ *        The domain setting the cookie
+ * @param string aReferrer
+ *        The referrer domain
+ * @param integer aLength
+ *        The count of cookies set/read during this event
+ * @param integer aMode
+ *        kMODE_SET_COOKIE or kMODE_GET_COOKIE
+ * @param integer aType
+ *        kTYPE_LOGIN_COOKIE or null
+ * @returns
```
Comment (monicachew) [inline lib/main.js]: @return the promise in case we need it later

File: lib/main.js [hunk 6]
```diff
@@ -180,12 +230,48 @@ function parseCookie(aCookie) {
...
+}
+
+/**
  * Log statistics for the given cookie.
  * @param kv {map} A map of cookie attributes that must include max-age.
  * @param domain {string} The domain setting the cookie.
  * @param referrerDomain {string} The top-level URL for third party cookies.
  */
-function logCookie(kv, domain, referrerDomain) {
+function originalLogCookie(kv, domain, referrerDomain) {
```
Comment (monicachew) [inline lib/main.js]: No need to keep this if we're not using it

**Comment by monicachew** [comment]
Awesome! I left you some comments, but I'd like to merge this in sooner than later 

----- Original Message -----

> A number of enhancements around storage data format and logging -
> still in progress
> You can merge this Pull Request by running
> git pull https://github.com/mozilla/cookiemonster storage-tweaks
> Or view, comment on, or merge it at:
> https://github.com/mozilla/cookiemonster/pull/26 Commit Summary
> - network monitor is up and working, began using monitor.record for
>   all new event types
> - cookie data is now recorded via monitor.record() instead of our
>   aggregated cookie object
> - pushing to add the xpi to the repo
>   File Changes
> - A cookiemonster.xpi (0)
> - M lib/main.js (302)
>   Patch Links:
> - https://github.com/mozilla/cookiemonster/pull/26.patch
> - https://github.com/mozilla/cookiemonster/pull/26.diff

File: lib/main.js (hunk 2 above)
Comment (daviddahl) [inline lib/main.js:96]: Ah. not with the added whitespace arg. I forgot about that.

File: lib/main.js (hunk 6 above)
Comment (daviddahl) [inline lib/main.js]: nope, kept it for reference. will remove.

File: lib/main.js (hunk 4 above)
Comment (daviddahl) [inline lib/main.js:514]: np. it is how we started the original web console.

File: lib/main.js (hunk 1 above)
Comment (daviddahl) [inline lib/main.js:554]: d'oh! nice.

**Comment by monicachew** [comment]
Ready for another look?

**Comment by daviddahl** [comment]
sure;) Sorry I got distracted by W3 mail / discussion today

**Comment by monicachew** [comment]
In that case, I'll go ahead and merge since I'd like to work on this tomorrow :)

**Pull request merged**